### DISCUSSION TITLE
The Competitive Exclusion Principle

### DISCUSSION THREAD

**Discussion by josePereiro** [josePereiro/TankeFactory#9] [Discussions]
Just a little discussion about a key principle in ecology/evolution. Here is a short historical paper about it: [The Competitive Exclusion Principle](https://doi.org/10.1126/science.131.3409.1292).

**Comment by josePereiro** [comment]
This principle is a good example of the kind of problems evolution-related theories have... There are a lot of undefined concepts even if the principles are really common-sensicals. 

I liked how it is stated here:

> "The most important lesson to be learned from evolutionary theory," says Michael Scriven in a brilliant essay recently published (26), "is a negative one: the theory shows us what scientific explanations need not do. In particular it shows us that one cannot regard explanations as unsatisfactory when they are not such as to enable the event in question to have been predicted." The theory of evolution is not one with which we can predict exactly the future course of species formation and extinction; rather, the theory "explains" the past. Strangely enough, we take mental satisfaction in this ex post facto explanation. Scriven has done well in showing why we are satisfied.

**Reply by ta-rekt** [reply]
This is exciting. The competitive exclusion principle seems to be the other side of the same coin as the paper I sent you, "the evolution of cooperation". Here it says that two species cannot compete for the exact same resources in the exact same environment, because one of them will inevitably grow at a faster rate and outcompete the other.

My first thought was: what if the growth rates are not fixed, but depend on the other species' size? For example, A and B are competing in identical environments. Initially, A's growth rate is $r_A$, B's growth rate is $r_B$, and their proportions are equal. But what if A's growth rate decreases proportionally to B's growth, and B's growth rate increases proportionally to A's decrease? Something like this:

$$\frac{dA}{dt} = r_a A$$

$$\frac{dB}{dt} = r_b B$$

$$\frac{dr_a}{dt} = - c_{ab} B$$

$$\frac{dr_b}{dt} = c_{ba} A $$

There should be a stable fixed point at which both growth rates are exactly equal, and the two species coexist. But I guess this situation is ruled out by the assumptions, since here the environment of the two species is not identical. The environment of A contains B, and vice versa. But isn't it unrealistic to suppose that two species are not contained in each other's environments?

This is a zero-sum game. The other side of the coin is games that are not zero-sum, which in the terminology of the Hardin paper you sent, means "ecological differentiation", or not competing for the same resource. In other words, there is a win-win situation if we cooperate.

**Reply by ta-rekt** [reply]
I want to try to prove that this set of equations has a stable fixed point. Do you have any idea how?

**Reply by josePereiro** [reply]
> I want to try to prove that this set of equations has a stable fixed point. Do you have any idea how?

Jajaj, I appreciate your faith in me. I understand the intuition, but I think the equations doesn't capture it...  For instance, $\frac{dr_a}{dt}$ is always negative or zero  (assuming $B \ge 0$). with $\frac{dr_b}{dt}$ it is the opposite. There is no feedback... At $t$ sufficiently large, $A$ will never increase and $B$ will never decrease

This is no proof, but this is a numerical simulation supporting what I say
<img width="709" alt="image" src="https://github.com/josePereiro/TankeFactory/assets/44244693/cb88f735-9ea8-4ae1-955b-7d3f097209c6">

**Reply by ta-rekt** [reply]
Okay, these equations have feedback:

$$\frac{dA}{dt} = r_a (B - A)$$

$$\frac{dB}{dt} = r_b (A - B)$$

![converging_populations](https://github.com/josePereiro/TankeFactory/assets/23706326/aacb8d62-bf80-4f56-9d92-b7a9c8212692)

Why don't these violate the competitive exclusion principle?

**Reply by josePereiro** [reply]
> Why don't these violate the competitive exclusion principle? 

Mmm, I think because the resources (lets define them as what is positively related with the population growth) in this case are each other's populations. This looks like a mutual depredation model. $A$ feeds from $B$ and $B$ feeds from $A$... The negative part is just a 'natural' death rate (which in this case affects the growth rate in the same proportion as the resource).

> The Competitive Exclusion Principle in ecology states that two species competing for the same limited resources cannot coexist indefinitely if other ecological factors remain constant.

I think this is equivalent to the follow equations (from the top of my head):

$$
\frac{dA}{dt} = f_A(R,A) + h_A(A)
$$

$$
\frac{dB}{dt} = f_B(R,B) + h_B(B)
$$

where $h_X$ at must depends just on the $X$'s population to comply with "_if other ecological factors remain constant_" (maybe this can be relax/generalized further). I also think that for $R$ to be considered a limiting resource $\frac{d}{dR} \frac{dX}{dt} > 0$ must be true for any growing population $X$. That is, that the growth rate positively depends on $R$. Otherwise it makes no sense to call it nor limiting nor a resource. That will constraint $\frac{df}{dR} > 0$ as well.

SUMMARY: The definition of resource is important and the definition of limiting too. Of course, a big problem testing the generality of the principle is at identifying if the conditions are meet in real systems. The "_if other ecological factors remain constant_" will be the obvious scapegoat... or that an extra resource is missing in the model...
